src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -1127,16 +1149,32 @@ public OutColumn(Vector<float> input, string modelFile)
             {
                 Input = input;
             }
+
+            public OutColumn(Vector<float> input, TensorFlowModelInfo tensorFlowModelContext)
+                : base(new Reconciler(tensorFlowModelContext), input)
+            {
+                Input = input;
+            }
         }
 
         private sealed class Reconciler : EstimatorReconciler
         {
             private readonly string _modelFile;
+            private readonly TensorFlowModelInfo _tensorFlowModelContext;
 
             public Reconciler(string modelFile)
             {
                 Contracts.AssertNonEmpty(modelFile);
                 _modelFile = modelFile;
```

Comment:
>_modelFile [](start = 16, length = 10)

I would probably get rid of _modelFile and just use _tensorFlowModelContext since you can create it from modelFile anyway.
Less complicated I think from code standpoint

Reply:
Hmm...I tried and think that its better to keep it as-is. The reason is because `TensorFlowUtils.LoadTensorFlowModel` requires `IHostEnvironment` which is not available until `Reconcile` method is hit.
Let me know if you are thinking of some other solution?

---
In reply to: [223836239](https://github.com/dotnet/machinelearning/pull/1191#discussion_r223836239) [](ancestors = 223836239)